src/PayID/xrp-pay-id-client.ts

```diff
@@ -23,6 +26,19 @@ export default class XRPPayIDClient extends PayIDClient {
    */
   async xrpAddressForPayID(payID: string): Promise<string> {
     const result = await super.addressForPayID(payID)
-    return result.address
+
+    const { address } = result
+    if (Utils.isValidXAddress(result.address)) {
+      return address
+    }
+    const isTest = !(this.network === XRPLNetwork.Main)
```

Comment:
seems reasonable - do you typically prefer to negate the whole statement instead of testing for inequality?

Reply:
nah, you're probably right and my brain is just too tired to have done this right the first time around. 

fixed.